### PR TITLE
fix: plugin first-time-run issues [ROAD-731]

### PR DESCRIPTION
**Pull request by ArtsiomCh**
- Honoring user's product selection on welcome screen before first ever scan
- Proper message when Code product is disabled due to unreachable org settings on backend
- Auth screen has priority to be shown in case of invalid/empty token
- Code scan endpoint url was internally corrupted after any Snyk settings change

chore: tests refactoring and improvement, also do only one actual CLI download test and use mocks in other tests

- [x] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated
- [x] README.md updated, if user-facing

### CODE REVIEW

File: src/integTest/kotlin/io/snyk/plugin/services/download/CliDownloaderServiceIntegTest.kt [hunk 1]
```diff
@@ -61,25 +62,31 @@ class CliDownloaderServiceIntegTest : LightPlatformTestCase() {
         assertTrue(latestReleaseInfo.tagName.isNotEmpty())
     }
 
+    /**
+     * Should be THE ONLY test where we actually do download the CLI
+     * !!! Do __MOCK__ cli download in ANY other test to reduce testing time needed !!!
+     */
```
Comment (bastiandoetsch) [inline src/integTest/kotlin/io/snyk/plugin/services/download/CliDownloaderServiceIntegTest.kt:68]: YES! 💜

Comment (bastiandoetsch) [inline src/integTest/kotlin/io/snyk/plugin/services/download/CliDownloaderServiceIntegTest.kt:68]: YES! 💜

File: src/main/kotlin/io/snyk/plugin/ui/UIUtils.kt [hunk 2]
```diff
@@ -84,12 +84,16 @@ fun insertTitleAndResizableTextIntoPanelColumns(
...
-    (pluginSettings().sastOnServerEnabled ?: true && pluginSettings().localCodeEngineEnabled ?: true) ->
-        " (disabled due to Local Code Engine)"
-    !(pluginSettings().sastOnServerEnabled ?: false) -> " (disabled in Snyk.io)"
-    else -> ""
+fun snykCodeAvailabilityPostfix(): String {
+    val sastOnServerEnabled = pluginSettings().sastOnServerEnabled
+    val localCodeEngineEnabled = pluginSettings().localCodeEngineEnabled
+    return when {
+        !isSnykCodeAvailable(pluginSettings().customEndpointUrl) -> " (disabled for endpoint)"
+        (sastOnServerEnabled == null && localCodeEngineEnabled == null) -> " (disabled due to unreachable server settings)"
```
Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/ui/UIUtils.kt:92]: better `undefined` settings?

Comment (bastiandoetsch) [inline src/main/kotlin/io/snyk/plugin/ui/UIUtils.kt:92]: better `undefined` settings?

**Review by bastiandoetsch — APPROVED** [review]
LGTM, minor stuff mentioned in comment.

File: src/main/kotlin/io/snyk/plugin/ui/UIUtils.kt (hunk 2 above)
Comment (ArtsiomCh) [reply src/main/kotlin/io/snyk/plugin/ui/UIUtils.kt:92]: Unreachable is a bit more correct word here. As undefined could mean we do reach server but such settings been undefined there, which is separate case.

**Pull request merged**